src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs

```diff
@@ -758,14 +767,48 @@ internal void ThrowPending()
 
         internal protected virtual Task OnCloseAsync(TimeSpan timeout)
         {
-            Contract.Requires(false, "OnCloseAsync needs to be implemented on derived classes");
```

Comment:
Reviewers: for the record, this Contract.Requires was originally intended to alert us wherever we failed to override OnXXXAsync.  But apparently this does not fire, even in Debug builds.  A Contract.Assert likely would have fired.  Because it really is a serious issue if we don't override, I converted this to a hard faulted Task.